src/java.base/share/classes/java/lang/foreign/SymbolLookup.java

```diff
@@ -292,6 +294,9 @@ static SymbolLookup libraryLookup(String name, Arena arena) {
     static SymbolLookup libraryLookup(Path path, Arena arena) {
         Reflection.ensureNativeAccess(Reflection.getCallerClass(),
                 SymbolLookup.class, "libraryLookup");
+        if (!(path.getFileSystem() == FileSystems.getDefault())) {
```

Comment:
Shouldn't this be written as `path.getFileSystem() != FileSystems.getDefault()` ?